src/auth/screens/auth-profile.screen.js

```diff
@@ -143,6 +147,38 @@ class AuthProfile extends Component {
                 />
               )}
             </SectionList>
+
+            <SectionList title="Sign Out">
+              <ListItem
+                title="Sign Out"
+                titleStyle={styles.logoutTitle}
+                leftIcon={{
+                  name: 'sign-out',
+                  color: colors.grey,
+                  type: 'octicon',
+                }}
+                onPress={() => {
+                  Alert.alert(
+                    'Sign Out',
+                    'Are you sure you want to sign out?',
+                    [
+                      {
+                        text: 'No',
+                      },
+                      {
+                        text: 'Yes',
+                        style: 'destructive',
+                        onPress: () => {
+                          // persistStore(configureStore).purge();
```

Comment:
Similarly will like to have this abstracted in a separate function (`clearStoreAndSignOut` for example)

So this is a little trickier then I thought it would be. So if we try purging the store - that still doesn't mean our local Redux state will be cleared. The same with `AsyncStorage.clear()`

First and foremost, AsyncStorage.clear should return a promise so I thought something like this would work:

```
clearStoreAndSignOut = () => {
  AsyncStorage.clear(() => {
    this.props.navigation.navigate('Login');
  });
}
```

However because that only clears AsyncStorage and not the actual current data, you'll notice that it navigates to `Login` but gets navigated right back to the `Main` tab screens because of how we [check our authentication state](https://github.com/gitpoint/git-point/blob/master/src/auth/screens/login.screen.js#L88) when that component mounts.

So what I think we can do is create a new action creator in `auth.action.js` for `LOGOUT`. In here we can do two things:

1. Clear async storage
2.  Reset the state of the current store [just like this](https://stackoverflow.com/questions/35622588/how-to-reset-the-state-of-a-redux-store/35641992#35641992)

Let me know what you think (and don't hesitate to ask for help in the gitter channel - it's what I'm here for :))